kuma/core/middleware.py

```diff
@@ -35,7 +36,10 @@ def process_request(self, request):
             new_path = prefixer.fix(prefixer.shortened_path)
             query = dict((smart_str(k), v) for
                          k, v in request.GET.iteritems() if k != 'lang')
-            return HttpResponsePermanentRedirect(urlparams(new_path, **query))
+
+            # Never use HttpResponsePermanentRedirect here.
+            # Its a temporary redirect and should return with http 302, not 301
```

Comment:
Update: it looks like ``?lang=de`` is used in the footer language selector when JS is disabled. And, because JS is disabled, it doesn't show up in Google Analytics 😄